Basket-Donation-Payments/Basket-Donation-Payments/View Controllers/Payment Flow/PaymentFormViewController.m

```diff
@@ -0,0 +1,144 @@
+    self.cardTextField = cardTextField;
+    UIButton *button = [UIButton buttonWithType:UIButtonTypeCustom];
+    button.layer.cornerRadius = 5;
+    button.backgroundColor = [UIColor systemBlueColor];
+    button.titleLabel.font = [UIFont systemFontOfSize:22];
+    [button setTitle:@"Pay" forState:UIControlStateNormal];
+    [button addTarget:self action:@selector(pay) forControlEvents:UIControlEventTouchUpInside];
+    self.payButton = button;
+    UIStackView *stackView = [[UIStackView alloc] initWithArrangedSubviews:@[cardTextField, button]];
+    stackView.axis = UILayoutConstraintAxisVertical;
+    stackView.translatesAutoresizingMaskIntoConstraints = NO;
+    stackView.spacing = 20;
+    [self.view addSubview:stackView];
+    [NSLayoutConstraint activateConstraints:@[
+        [stackView.leftAnchor constraintEqualToSystemSpacingAfterAnchor:self.view.leftAnchor multiplier:2],
+        [self.view.rightAnchor constraintEqualToSystemSpacingAfterAnchor:stackView.rightAnchor multiplier:2],
+        [stackView.topAnchor constraintEqualToSystemSpacingBelowAnchor:self.view.safeAreaLayoutGuide.topAnchor multiplier:2],
+    ]];
+    
+    [self createPaymentIntent];
+    
+}
+
+- (void)createPaymentIntent {
+    NSString *backendURL = [APIManager getAPISecretKeysDict][@"Backend_Server_Url"];
+    // Create a PaymentIntent by calling your server's endpoint.
+    NSURL *url = [NSURL URLWithString:[NSString stringWithFormat:@"%@create-payment-intent", backendURL]];
+    NSDictionary *json = @{
+        @"items": @[
+                @{@"merchantId1": @"abc",
+                  @"price": @8000
+                },
+                @{@"merchantId2": @"cde",
+                  @"price": @8000
+                },
+                @{@"merchantId3": @"efg",
+                  @"price": @1000 //in cents
+                }
+        ],
+    };
+    NSData *body = [NSJSONSerialization dataWithJSONObject:json options:0 error:nil];
+    NSMutableURLRequest *request = [[NSURLRequest requestWithURL:url] mutableCopy];
+    [request setHTTPMethod:@"POST"];
+    [request setValue:@"application/json" forHTTPHeaderField:@"Content-Type"];
+    [request setHTTPBody:body];
+    NSURLSessionTask *task = [[NSURLSession sharedSession] dataTaskWithRequest:request completionHandler:^(NSData *data, NSURLResponse *response, NSError *requestError) {
+        NSError *error = requestError;
+        NSHTTPURLResponse *httpResponse = (NSHTTPURLResponse *)response;
+        if (error != nil || httpResponse.statusCode != 200) {
+            UIAlertController *alert = [Utils createAlertControllerWithTitle:@"Error loading page." andMessage:error.localizedDescription okCompletion:nil cancelCompletion:nil];
+            [self presentViewController:alert animated:YES completion:nil];
+        }
+        else {
+            NSDictionary *dataDict =[NSJSONSerialization JSONObjectWithData:data options:0 error:nil];
+            NSLog(@"Created PaymentIntent %@", dataDict[@"description"]);
```

Comment:
Remember to clean this debug logs later, specially anything that could leak sensitive user information, the console logs of the device are not secure.